### PR TITLE
Adds veggieseasons preferences, tweaks styles + list page

### PR DESCRIPTION
**Pull request by RedBrogdon**
This is the first of several PRs that will bring in changes from this branch:

https://github.com/RedBrogdon/samples/tree/settings/veggieseasons

Included are the new model for preferences used by the settings screen, some new styling constants, a tweak to the "season circles" used on the details page, the Navigator-related fix we talked about offline, and a bunch of auto-generated project files changes.

Future PRs will include use of our new veggie photos (probably in a separate repo and package), cards for the list screen, modal display of the detail screen, and a few other things.

### CODE REVIEW

File: veggieseasons/lib/screens/details.dart [hunk 1]
```diff
@@ -13,9 +13,10 @@ import 'package:veggieseasons/styles.dart';
 /// harvested. It displays the first two letters of the season and uses a
 /// different background color to represent each of the seasons as well.
 class SeasonCircle extends StatelessWidget {
+  final bool active;
```
Comment (xster) [inline veggieseasons/lib/screens/details.dart]: dartdoc it. Just looking at it, I can't tell if it's the current season or the focused season.

File: veggieseasons/lib/screens/details.dart [hunk 2]
```diff
@@ -116,8 +121,12 @@ class DetailsScreen extends StatelessWidget {
             mainAxisSize: MainAxisSize.max,
             children: [
               Wrap(
-                children:
-                    veggie.seasons.map<Widget>((s) => SeasonCircle(s)).toList(),
+                children: Season.values
+                    .map((s) => SeasonCircle(
```
Comment (xster) [inline veggieseasons/lib/screens/details.dart]: multilined => with indentations etc are hard to read. Use {}

File: veggieseasons/lib/screens/details.dart [hunk 3]
```diff
@@ -116,8 +121,12 @@ class DetailsScreen extends StatelessWidget {
...
+                          s,
+                          veggie.seasons.contains(s),
+                        ))
+                    .toList(),
```
Comment (xster) [inline veggieseasons/lib/screens/details.dart]: The indents are a bit hard to read

File: veggieseasons/lib/screens/details.dart (hunk 1 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/details.dart]: Renamed and better explained.

File: veggieseasons/lib/data/preferences.dart [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  static const caloriesKey = 'calories';
+  static const preferredCategoriesKey = 'preferredCategories';
+
+  int _desiredCalories = 2000;
+
+  int get desiredCalories => _desiredCalories;
+
+  Set<VeggieCategory> _preferredCategories = Set<VeggieCategory>();
+
+  Set<VeggieCategory> get preferredCategories => Set.from(_preferredCategories);
```
[the remark this answers is not in the document]
Comment (RedBrogdon) [reply veggieseasons/lib/data/preferences.dart]: Should have thought about this bit longer.

Added a Future<void> to keep track of the call to _load, which the getters now await.

File: veggieseasons/lib/screens/details.dart (hunk 2 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/details.dart]: Done.

File: veggieseasons/lib/screens/details.dart (hunk 3 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/details.dart]: Fixed.

**Comment by xster** [comment]
LGTM

**Pull request merged**